app/main.py

```diff
@@ -1,6 +1,23 @@
-from typing import Callable
+from typing import Callable, Any, Dict, Tuple
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict: Dict[str, Dict[Tuple, Any]] = {}  # Общий кэш для всех функций
+
+    def wrapper(*args, **kwargs):
+        key = (args, tuple(kwargs.items()))
```

Comment:
The key generation uses `tuple(kwargs.items())`, which may not work as expected if the order of keyword arguments changes. Consider using `frozenset(kwargs.items())` to ensure the key is order-independent.